src/Toolbox.php

```diff
@@ -1645,7 +1645,9 @@ public static function manageRedirect($where)
                     Html::redirect($redirect_to);
                 }
 
-                $data = explode("_", $where);
+                // explode with limit 3 to preserve the last part of the url
+                // /index.php?redirect=ticket_2_Ticket$main#TicketValidation_1 (preserve anchor)
+                $data = explode("_", $where, 3);
```

Comment:
Are we sure 3 will always be enough? What about exploding on `#` then append it back when/where needed?

Reply:
Yes, explode ```_``` is only used to find the ticket ID, 

and originally we didn't go any further than index ```2``` (so ```3``` part)